parlai/core/torch_ranker_agent.py

```diff
@@ -73,8 +73,9 @@ def __init__(self, opt, shared=None):
             self.model.cuda()
             self.rank_loss.cuda()
 
-        optim_params = [p for p in self.model.parameters() if p.requires_grad]
-        self.init_optim(optim_params)
+        if not shared:
+            optim_params = [p for p in self.model.parameters() if p.requires_grad]
```

Comment:
:pikachu_meme:

Reply:
I believe optimizer needs to come from shared or something though, hence why it's breaking unit tests